### ISSUE TITLE
Should we switch from ELB to NLBs in AWS?

### ISSUE THREAD

**Issue by rfranzke** [gardener/gardener#242]
AWS supports network load balancers (https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html) which are [cheaper](https://aws.amazon.com/elasticloadbalancing/pricing/) and provide more features.

Should we switch from the elastic load balancers which we use today to network load balancers?

**Comment by vlerenc** [comment]
Yes, I believe so. When I requested the limit raise for our accounts, I was anyways informed, that CLBs are on their way out and we should rather use NLBs. Unless we find any showstoppers, already the price speaks for NLBs.

**Comment by vlerenc** [comment]
What is the price difference?

**Comment by harsimranmaan** [comment]
Can I take this?

**Comment by rfranzke** [comment]
Sure, that would be great @harsimranmaan! Thank you!

**Comment by rfranzke** [comment]
Hi @harsimranmaan, do you have any update on this one?

**Comment by rfranzke** [comment]
NLBs preserve the client IPs and don't have dedicated security groups. Instead, the security groups attached to the underlying instances controls the traffic.
Currently, we don't allow arbitrary traffic to the nodes, making NLBs unable to work.

A manual workaround is to add a new rule
```
Custom TCP Rule
TCP
30000 - 32767
0.0.0.0/0
```
to our existing nodes security group. However, we probably want to enable NLBs out-of-the-box. What's the best way doing this?

More information:
* https://forums.aws.amazon.com/thread.jspa?threadID=263245
* https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-register-targets.html#target-security-groups

**Comment by vlerenc** [comment]
Whoa, that's bad, thank you! This may also explain another incident reported with NLBs in the last days.
You picked this port range, because it's the NodePort range? Yes, something like this seems sensible. What other options would you see?

**Comment by rfranzke** [comment]
> You picked this port range, because it's the NodePort range?
Right.

> What other options would you see?
Not much other options, but I haven't spend time to check it more thoroughly.
Another option could be to do nothing and leave it to the customer to enrich the security group with a rule that allows only the desired client IPs (instead of 0.0.0.0/0).

All our worker nodes don't have public IPs. If we would go and add 0.0.0.0/0 by default then nobody from outside could contact the nodes, but everybody that has a route to the worker subnet can. WDYT?

**Comment by vlerenc** [comment]
Sure, don't see many other alternatives besides not doing anything. ;-)
